### PR TITLE
Some small changes to the interface

### PR DESCRIPTION
**Pull request by RadStr**

### CODE REVIEW

**Comment by RadStr** [comment]
I added some comments on the all things I could think of, some more could come once I actually implement something.

You also said that you also had some notes, so that could also ignite some more ideas.

**Review by RadStr — COMMENTED** [review]
Oops I didn't know I had to finish the review, ok so the comments should be online now.

File: src/diagram/diagram-api.tsx [hunk 1]
```diff
@@ -12,6 +12,8 @@ export interface DiagramActions {
 
   removeGroups(groups: string[]): void;
 
+  extendGroup(group: Group, extension: string[]): void;
```
[the remark this answers is not in the document]
Comment (skodapetr) [reply src/diagram/diagram-api.tsx:15]: Just a question why not have setGroup directly? Do we get benefit fow having an extra method here? I would assume the implementation would be similar and we probably want the "owner" to know groups any way.

File: src/diagram/diagram-api.tsx [hunk 2]
```diff
@@ -32,7 +34,9 @@ export interface DiagramActions {
 
   updateEdges(edge: Edge[]): void;
 
-  updateEdgesWaypointPosition(positions: { [identifier: string]: { [identifier: string]: Position } }): void;
+  updateEdgesWaypointPosition(positions: { [identifier: string]: Position[] }): void;
+
+  straightenEdges(identifier: string[]): void;
```
[the remark this answers is not in the document]
Comment (skodapetr) [reply src/diagram/diagram-api.tsx:39]: What does "straightenEdges" means? Just remove the edges? If so, I would stick with the updateWaypoints, where we provide empty array.

File: src/diagram/diagram-api.tsx
[the remark this answers is not in the document]
Comment (skodapetr) [reply src/diagram/diagram-api.tsx:1]: 1. No, I would keep startedPosition to be the original position. When user change the node/waypoint position we keep the chenge in the editor only. When needed we can query the editor for the positions. 
Should we need collaborative edit, we can a callback anouncing change in a position, probably with some debounce.

2. We can change view to viewport, that may be actually more representative. Good point.

3. Again, good question. What about changing them to initialWaypoints? The idea is that we want to keep this position related information in the editor as much as possible.

File: src/diagram/diagram-api.tsx [hunk 3]
```diff
@@ -32,7 +34,9 @@ export interface DiagramActions {
 
   updateEdges(edge: Edge[]): void;
 
-  updateEdgesWaypointPosition(positions: { [identifier: string]: { [identifier: string]: Position } }): void;
+  updateEdgesWaypointPosition(positions: { [identifier: string]: Position[] }): void;
```
[the remark this answers is not in the document]
Comment (skodapetr) [reply src/diagram/diagram-api.tsx:37]: `setEdgesWaypointPosition` is definitely better name. About the identifier, you are right we can keep it as an array.

**Comment by skodapetr** [comment]
@RadStr  Can you update the PR?

File: src/diagram/diagram-api.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (RadStr) [reply src/diagram/diagram-api.tsx:15]: I agree setGroup is better.

File: src/diagram/diagram-api.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (RadStr) [reply src/diagram/diagram-api.tsx:39]: Yes, just remove the waypoints. At least I assume that you meant waypoints and not edges. I agree that it is unnecessary method, but it is nice to have. So just a question, currently we are providing in the API just the necessary methods, i.e. have the API lean as possible?

**Comment by RadStr** [comment]
> @RadStr Can you update the PR?

Sure

File: src/diagram/diagram-api.tsx [hunk 4]
```diff
@@ -8,11 +8,13 @@ export interface DiagramActions {
...
 
-  addGroup(group: Group): void;
+  addGroup(group: Group, content?: string[]): void;
 
   removeGroups(groups: string[]): void;
 
-  extendGroup(group: Group, extension: string[]): void;
+  setGroup(group: Group, content: string[]): void;
+
+  getGroupContent(group: Group): string[];
```
Comment (RadStr) [inline src/diagram/diagram-api.tsx:17]: Add new getGroupContent because currently there was no way to query the group content outside of editor.

**Review by RadStr — COMMENTED** [review]
Changed the API based on feedback and added 2 new comments

File: src/diagram/diagram-api.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (skodapetr) [reply src/diagram/diagram-api.tsx]: From my perspective:
Yes, API should be minimal to hide as much internals as possible and simplify the contract between the API producer and consumer.

We can provide a helper with the API. This helper can be created and maintained by the consumer and later, when API is expanded, for example to address performance issues, the helper can be re-implemented internally to leverage that.

**Pull request merged**